src/conn/conn_tiered.c

```diff
@@ -175,32 +175,49 @@ __tiered_manager_config(WT_SESSION_IMPL *session, const char **cfg, bool *runp)
  *     Parse and setup the storage server options.
  */
 static int
-__tiered_config(WT_SESSION_IMPL *session, const char **cfg, bool *runp)
+__tiered_config(WT_SESSION_IMPL *session, const char **cfg, bool *runp, bool reconfig)
 {
-    WT_CONFIG_ITEM cval;
+    WT_CONFIG_ITEM bucket, cval;
     WT_CONNECTION_IMPL *conn;
-    bool enabled;
+    WT_DECL_RET;
 
     conn = S2C(session);
 
-    WT_RET(__wt_config_gets(session, cfg, "tiered_storage.enabled", &cval));
-    enabled = cval.val != 0;
+    if (!reconfig) {
+        WT_RET(__wt_config_gets(session, cfg, "tiered_storage.name", &cval));
+        WT_RET(__wt_config_gets(session, cfg, "tiered_storage.bucket", &bucket));
+        WT_RET(__wt_tiered_bucket_config(session, &cval, &bucket, &conn->bstorage));
+    }
```

Comment:
I have to check if we can get here via reconfig with a `cfg` stack containing the default settings.

Reply:
Actually @ddanderson if we have a table with a different storage source or bucket than the system one, how should the user update the `kmsid`? What API would they use? Maybe they would use `conn->reconfigure` to specify the name and bucket to identify the existing entry?

Also, is `kmsid` too specific? Do we want the more generic `auth_token`? I don't have a strong feeling either way.

Reply:
@sueloverso I assume that if we have a table with a different storage source/bucket, then wouldn't you need to indicate  the storage source name, bucket name and `kmsid` when you create the table?

I agree with you that `kmsid` is too specific.  I think we should start using `auth_token` everywhere.

Reply:
Ok. I'll revert `kmsid` to `auth_token` everywhere.

If someone does `wiredtiger_open("tiered_storage=(name=s3,bucket=mybucket,auth_token=token1)")` and then needs to update the token, `conn->reconfigure("tiered_storage=(auth_token=token1prime)")` will update the system storage.

But if someone also does `session->create(URI, "tiered_storage=(name=s3,bucket=other,auth_token=token2)")` there is no equivalent to update the `auth_token`. Hmm, maybe `session->alter` perhaps. That is the only thing we have.

Reply:
I don't believe it will ever need to be updated (at least in the atlas world).  The token is an entry into a key-vault (that is KMS).  If keys need to be rotated or updated, it happens inside the "vault", and the token remains the same.  At least that is my understanding.